src/test/java/com/internship/backend/controller/AuthorityControllerTest.java

```diff
@@ -0,0 +1,110 @@
+package com.internship.backend.controller;
+
+import com.internship.backend.dto.AuthorityDTO;
+import com.internship.backend.exceptions.UserDoesNotExistException;
+import com.internship.backend.model.Authority;
+import com.internship.backend.service.AuthorityService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.util.List;
+
+import static org.hamcrest.Matchers.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+class AuthorityControllerTest {
+
+    @InjectMocks
+    private AuthorityController authorityController;
+
+    @Mock
+    private AuthorityService authorityService;
+
+    private MockMvc mockMvc;
+    private Authority authority;
+    private AuthorityDTO authorityDTO;
+    private List<Authority> authorities;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+        mockMvc = MockMvcBuilders.standaloneSetup(authorityController).build();
+
+        authority = Authority.builder()
+                .id(1)
+                .name("ROLE_USER")
+                .build();
+
+        authorityDTO = new AuthorityDTO("ROLE_USER", 1);
+
+        authorities = List.of(authority);
+    }
+
+    @Test
+    void createAuthority() throws Exception {
+        when(authorityService.add(any(AuthorityDTO.class))).thenReturn(authority);
+
+        mockMvc.perform(post("/api/authority/addAuthority/1")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("{\"name\":\"ROLE_USER\",\"userId\":1}"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.name", is(authority.getName())));
+
+        verify(authorityService, times(1)).add(any(AuthorityDTO.class));
+    }
+
+    @Test
+    void createAuthorityThrowsBadRequest() throws Exception {
+        when(authorityService.add(any(AuthorityDTO.class))).thenThrow(UserDoesNotExistException.class);
+
+        mockMvc.perform(post("/api/authority/addAuthority/1")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("{\"name\":\"ROLE_USER\",\"userId\":1}"))
+                .andExpect(status().isBadRequest());
+
+        verify(authorityService, times(1)).add(any(AuthorityDTO.class));
+    }
+
+    @Test
+    void getAllAuthorities() throws Exception {
```

Comment:
In general a test name should be meaningful and somehow explaining what is the purpose of the test. In this case it will be to test that getAllAuthorities method actually works as expected. In my opinion having the test name equal to the method tested is not a good idea, since it not provide details about what is expected from the test. 
https://jonasg.io/posts/subtle-art-of-java-test-method-naming/

Reply:
As an alternative I think that prefixing with "should" and adding short explaining details about what should be expected from the tests will be useful here. For this test in particular it will be something like: shouldRetrieveAllAuthorities(). Same for other tests names.